### PR TITLE
No google api key should raise an error only if no client/secret provided

### PR DESCRIPTION
**Pull request by ebreton**

### CODE REVIEW

File: geocoder/google.py [hunk 1]
```diff
@@ -182,7 +188,10 @@ def _build_params(self, location, provider_key, **kwargs):
             params['client'] = self.client
             return self._encode_params(params)
         # or API key
+        # In this case: raise exception if not valid key found
         else:
+            if not provider_key:
+                raise ValueError('Provide API Key')
```
Comment (DenisCarriere) [inline geocoder/google.py]: Wouldn't this raise an Error if you **don't** provide an API key?

Using the Google Provider an API key is optional

**Comment by DenisCarriere** [comment]
Yep it crashes when you don't provide an API key.

```bash
$ export GOOGLE_API_KEY=''
$ python google.py        
Traceback (most recent call last):
  File "google.py", line 288, in <module>
    g = GoogleQuery('11 Wall Street, New York')
  File "/Users/mac/Github/geocoder/geocoder/base.py", line 724, in __init__
    'params', self._build_params(location, provider_key, **kwargs)))
  File "google.py", line 194, in _build_params
    raise ValueError('Provide API Key')
ValueError: Provide API Key
```

**Comment by DenisCarriere** [comment]
@ebreton Updated base.py

File: geocoder/google.py (hunk 1 above)
Comment (ebreton) [inline geocoder/google.py]: Ahh... I thought that you would need either a CLIENT or an API KEY.

but from what your example, we could have neither of them ?

**Pull request merged**